usaspending_api/etl/swap_in_new_table_strategies.py

```diff
@@ -0,0 +1,955 @@
+                swap in table.
+            django_command: An instance of a Django command that will be using this strategy.
+        """
+        self._options = swap_table_options
+        self._django_command = django_command
+
+    def swap_in_new_table(
+        self,
+    ):
+        if self._load_tracker is None:
+            raise ValueError("LoadTracker property of this istance must be set before invoking swap in new table.")
+
+        self._validate_options()
+
+        table = self._options["table"]
+        # TODO: PIPE-513 developed a solution that forces this knowledge duplication hopefully that other implementation can be fixed in the future.
+        # Fixing that other implementation would enable the suite of swap_in_new_table classes to not duplicate knowledge.
+        upsert_table_name = f"{table}_temp_upserts"
+        delete_table_name = f"{table}_temp_deletes"
+        temp_table_list = [upsert_table_name, delete_table_name]
+        with connection.cursor() as cursor:
+            # Go ahead and retrieve the schema for both table; used for some validation checks and the final swap
+            cursor.execute(
+                f"""
+                SELECT tablename, schemaname
+                FROM pg_tables
+                WHERE tablename = '{table}' OR tablename = '{delete_table_name}' OR tablename = '{upsert_table_name}'
+                UNION
+                SELECT matviewname AS tablename, schemaname
+                FROM pg_matviews
+                WHERE matviewname = '{table}' OR tablename = '{delete_table_name}' OR tablename = '{upsert_table_name}'
+            """
+            )
+            schemas_lookup = {table: schema for (table, schema) in cursor.fetchall()}
+
+            # Begin fetching schemas of the tables involved in this operation
+            curr_schema_name = schemas_lookup.get(table)
+            qualified_dest_table = f"{curr_schema_name}.{table}"
+            temp_schema_name = schemas_lookup.get(delete_table_name)
+
+            # We assume both temp tables have the same schema, but let's double check
+            if temp_schema_name != schemas_lookup.get(upsert_table_name):
+                raise ValueError(
+                    f"The temp tables by the names {delete_table_name} and {upsert_table_name} must reside in the same schema. Currently, they do not."
+                )
+
+            qualified_upsert_postgres_table = f"{temp_schema_name}.{upsert_table_name}"
+            qualified_delete_postgres_table = f"{temp_schema_name}.{delete_table_name}"
+            self._validate_live_tables(cursor, table)
+            self._validate_temp_tables(cursor, temp_table_list)
+
+            self._execute_upserts(cursor, qualified_dest_table, qualified_upsert_postgres_table)
+            self._execute_deletes(cursor, qualified_dest_table, qualified_delete_postgres_table)
+
+            # Begin refreshing views
```

Comment:
For regular views, no recreation is necessary because a view queries the source data at runtime.

Materialized views can simply be refreshed rather than recreated.

Reply:
addressed